Source/VesselData.cs

```diff
@@ -1,798 +1,890 @@
+                        if (Persistence == HighLogic.SaveFolder.ToString() || Persistence == "WhitecatsDummySaveFileThatNoOneShouldNameTheirSave")
+                        {
+                            VesselInformation.AddNode(vessel);
+                        }
+                    }
+                }
+            }
+            print("WhitecatIndustries - Orbital Decay - Loaded vessel data.");
+        }
+
+        public void FixedUpdate()
+        {
+            if (Time.timeSinceLevelLoad > 0.3)
+            {
+                if ((Time.time - lastUpdate) > UPTInterval) // 1.4.0 Lag Busting
+                {
+                    lastUpdate = Time.time;
+
+                    if (HighLogic.LoadedSceneIsGame && (HighLogic.LoadedScene != GameScenes.LOADING && HighLogic.LoadedScene != GameScenes.LOADINGBUFFER && HighLogic.LoadedScene != GameScenes.MAINMENU))
+                    {
+                        Vessel vessel = new Vessel();
+                        for (int i = 0; i < FlightGlobals.Vessels.Count; i++)
+                        {
+                            vessel = FlightGlobals.Vessels.ElementAt(i);
+                            {
+                                if (CheckIfContained(vessel) == true)
+                                {
+                                    if (vessel.situation != Vessel.Situations.SUB_ORBITAL && vessel.situation != Vessel.Situations.ORBITING)
+                                    {
+                                        ClearVesselData(vessel);
+                                    }
+
+                                    if (vessel.situation == Vessel.Situations.ORBITING || vessel.situation == Vessel.Situations.SUB_ORBITAL)
+                                    {
+                                        WriteVesselData(vessel);
+                                    }
+                                }
+                                else if (CheckIfContained(vessel) == false)
+                                {
+                                    if (vessel.situation == Vessel.Situations.ORBITING || vessel.situation == Vessel.Situations.SUB_ORBITAL) // 1.4.2
+                                    {
+                                        WriteVesselData(vessel);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public void OnDestroy()
+        {
+            if (DecayManager.CheckSceneStateMain(HighLogic.LoadedScene))
+            {
+                if ((Planetarium.GetUniversalTime() == HighLogic.CurrentGame.UniversalTime) || HighLogic.LoadedScene == GameScenes.FLIGHT)
+                {
+                    print("WhitecatIndustries - Orbital Decay - Vessel Information saved. Ondestroy");
+                    File.ClearNodes();
+                   // VesselInformation.Save(FilePath);

```

Comment:
Thanks a lot for this Zajc3w I really appreciate the volume of work you have put into the code! 
Just going to note some points for myself when integrating the new code! 

- VesselInformation still requires saving for decay functions, will implement and test out asap